packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -519,17 +519,7 @@ void _validateFlutter(YamlMap? yaml, List<String> errors) {
           _validateFonts(yamlValue, errors);
         }
       case 'licenses':
-        if (yamlValue is! YamlList) {
-          errors.add('Expected "$yamlKey" to be a list of files, but got $yamlValue (${yamlValue.runtimeType})');
-        } else if (yamlValue.isEmpty) {
```

Comment:
Do we still need to check within this `case` if the `yamlValue.isEmpty` and break?

Reply:
Empty lists should pass the `_validateList` check, so I don't think so.